trade-finance-manager-ui/server/controllers/utilisation-reports/record-corrections/initiate-record-correction-request/index.ts

```diff
@@ -0,0 +1,108 @@
+ * @param reportId - The ID of the utilisation report.
+ * @param initiateRecordCorrectionRequestError - The error key to be set in the session.
+ * @param checkedCheckboxIds - An object representing the checked checkbox IDs.
+ * @returns The redirect response.
+ */
+const redirectToPremiumPaymentsWithError = (
+  req: PostInitiateRecordCorrectionRequest,
+  res: Response,
+  reportId: string,
+  initiateRecordCorrectionRequestError: InitiateRecordCorrectionRequestErrorKey,
+  checkedCheckboxIds: Record<string, true | undefined> = {},
+) => {
+  req.session.initiateRecordCorrectionRequestErrorKey = initiateRecordCorrectionRequestError;
+  req.session.checkedCheckboxIds = checkedCheckboxIds;
+
+  const premiumPaymentsFacilityId = getPremiumPaymentsFacilityIdQueryFromReferer(req.headers.referer);
+  return res.redirect(axios.getUri({ url: `/utilisation-reports/${reportId}`, params: { premiumPaymentsFacilityId } }));
+};
+
+/**
+ * Controller for the POST initiate record correction request route.
+ *
+ * Validates selected fee records for creating a record correction request.
+ *
+ * Redirects to the premium payments tab with the appropriate error message
+ * if the selections are not valid for record correction request creation.
+ *
+ * Redirects to the create record correction request page for the selected
+ * fee record if the selections are valid for record correction request creation.
+ *
+ * @param req - The request object
+ * @param res - The response object
+ */
+export const postInitiateRecordCorrectionRequest = (req: PostInitiateRecordCorrectionRequest, res: Response) => {
+  try {
+    const { reportId } = req.params;
+
+    const checkedCheckboxIds = getPremiumPaymentsCheckboxIdsFromObjectKeys(req.body);
+
+    if (checkedCheckboxIds.length === 0) {
+      return redirectToPremiumPaymentsWithError(
+        req,
+        res,
+        reportId,
+        INITIATE_RECORD_CORRECTION_ERROR_KEY.NO_FEE_RECORDS_SELECTED,
+        mapCheckedCheckboxesToRecord(checkedCheckboxIds),
+      );
+    }
+
+    if (checkedCheckboxIds.length > 1) {
+      return redirectToPremiumPaymentsWithError(
+        req,
+        res,
+        reportId,
+        INITIATE_RECORD_CORRECTION_ERROR_KEY.MULTIPLE_FEE_RECORDS_SELECTED,
+        mapCheckedCheckboxesToRecord(checkedCheckboxIds),
+      );
+    }
+
+    const checkedCheckboxId = checkedCheckboxIds[0];
+    const selectedCheckboxStatus = getFeeRecordStatusFromPremiumPaymentsCheckboxId(checkedCheckboxId);
+
+    if (selectedCheckboxStatus !== FEE_RECORD_STATUS.TO_DO) {
+      return redirectToPremiumPaymentsWithError(
+        req,
+        res,
+        reportId,
+        INITIATE_RECORD_CORRECTION_ERROR_KEY.INVALID_STATUS,
+        mapCheckedCheckboxesToRecord(checkedCheckboxIds),
+      );
+    }
+
+    const selectedFeeRecordIds = getFeeRecordIdsFromPremiumPaymentsCheckboxIds([checkedCheckboxId]);
+
+    if (selectedFeeRecordIds.length === 0 || selectedFeeRecordIds.length > 1) {
```

Comment:
Is there a specific reason the condition isn't just `selectedFeeRecordIds.length !== 1` ?

Reply:
Nope, just didn't occur to me 😅